helm-chart/templates/nats.yaml

```diff
@@ -16,6 +16,75 @@ metadata:
     app.kubernetes.io/part-of: nginx-service-mesh
 data:
   nats.conf: {{ .Files.Get "configs/nats.conf" | quote }}
+{{- if eq .Values.environment "openshift" }}
+---
+kind: SecurityContextConstraints
+metadata:
+  name: nginx-mesh-nats-permissions
+groups: []
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: false
+allowPrivilegedContainer: false
+allowedCapabilities:
+  - NET_ADMIN
+  - KILL
+apiVersion: security.openshift.io/v1
+defaultAddCapabilities: null
+fsGroup:
+  type: MustRunAs
+priority: null
+readOnlyRootFilesystem: false
+requiredDropCapabilities:
+  - ALL
+runAsUser:
+  type: MustRunAs
+  uid: 2102
+seLinuxContext:
+  type: MustRunAs
+supplementalGroups:
+  type: MustRunAs
+users: []
+volumes:
+  - secret
+  - csi
+  - configMap
+  - emptyDir
```

Comment:
nit: move all list indentations to the left for consistency (nested directly under the parent) for all resources you've added